### PR TITLE
Wrap http exceptions from requests lib with internal exceptions

### PR DESCRIPTION
**Pull request by abrahamko**
http_wrapper should not throw exceptions from its internal
library, `requests`.
Instead, it takes the necessary info from the exceptions and use it
to create new internal exception.
This will allow replacing `requests` lib without impacting code
outside of http_wrapper.

### Desired Outcome

*Please describe the desired outcome for this PR.  Said another way, what was
the original request that resulted in these code changes?  Feel free to copy
this information from the connected issue.*

### Implemented Changes

*Describe how the desired outcome above has been achieved with this PR. In
particular, consider:*

- _What's changed? Why were these changes made?_
- _How should the reviewer approach this PR, especially if manual tests are required?_
- _Are there relevant screenshots you can add to the PR description?_

### Connected Issue/Story

Resolves #[relevant GitHub issue(s), e.g. 76]

CyberArk internal issue link: [insert issue ID]()

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [ ] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [ ] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [ ] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

File: conjur/errors.py [hunk 1]
```diff
@@ -168,3 +168,24 @@ class KeyringWrapperSetError(KeyringWrapperGeneralError):
     def __init__(self, message: str = ""):
         self.message = message
         super().__init__(self.message)
+
+class HttpError(Exception):
+    """ Base exception for general HTTP failures """
+    def __init__(self, message: str = "HTTP request failed", response: str = ""):
+        self.message = message
+        self.response = response
```
Comment (mbenita-Cyberark) [inline conjur/errors.py:176]: I think you can omit the variables assignments since it happens in the base class
```suggestion
```

**Review by mbenita-Cyberark — COMMENTED** [review]
LGTM.
minor comment

File: conjur/errors.py (hunk 1 above)
Comment (abrahamko) [reply conjur/errors.py:176]: done

**Review by mbenita-Cyberark — APPROVED** [review]
Great

**Review by mbenita-Cyberark — APPROVED** [review]

**Pull request merged**